src/v/cluster/rm_stm.cc

```diff
@@ -1774,19 +1775,37 @@ ss::future<> rm_stm::do_try_abort_old_tx(model::producer_identity pid) {
 
     _mem_state.expected.erase(pid);
 
-    std::optional<prepare_marker> marker;
-    auto prepare_it = _mem_state.preparing.find(pid);
-    if (prepare_it != _mem_state.preparing.end()) {
-        marker = prepare_it->second;
+    std::optional<model::tx_seq> tx_seq;
+    if (!is_transaction_ga()) {
+        auto prepare_it = _mem_state.preparing.find(pid);
+        if (prepare_it != _mem_state.preparing.end()) {
+            tx_seq = prepare_it->second.tx_seq;
+        }
+        prepare_it = _log_state.prepared.find(pid);
+        if (prepare_it != _log_state.prepared.end()) {
```

Comment:
is it possible that pid is in both `_log_state.prepared` and `_mem_state.prepared` ? If that is the case the value will be overridden, is this expected ?

Reply:
It isn't a problem, we only need to discover id (txseq) of the ongoing tx: we used it before with `prepare_marker`